adapters/repos/db/vector/hnsw/search.go

```diff
@@ -441,14 +442,7 @@ func (h *hnsw) distanceToFloatNode(distancer distancer.Distancer,
 ) (float32, bool, error) {
 	candidateVec, err := h.vectorForID(context.Background(), nodeID)
 	if err != nil {
-		var e storobj.ErrNotFound
```

Comment:
Are we fine deleting this section? Don't we still have to call `handleDeletedNode` when the node is not in the cache?

Reply:
We do, but we handle the error where we have the information to take action. Here we are just hiding the reason of the error and the caller cannot handle it properly.